### ISSUE TITLE
documents4j-server-standalone-1.0.3-shaded.jar - is it independent related to OS

### ISSUE THREAD

**Issue by vinayajo** [documents4j/documents4j#53]
hi i have run as jar in windows  server os machine. every 3 hours command prompt automatically closed.  i dont know the root issue. i am planing to  deploy this in UAT or prod? how it is stable? any recommendations. please reply at earliest and advise

**Comment by raphw** [comment]
It should be stable. Have you looked into how to run the server as a service? It needs some setup.

**Comment by vinayajo** [comment]
i am running this as service java -jar documents4j-server-standalone-shaded.jar http://server:9998 by calling from command prompt

**Comment by raphw** [comment]
You can find instructions on the webpage: http://documents4j.com/#/

You need to run the server on a Windows machine as the server connects to MS Word via .NET APIs.

I assume that you trigger some timeout on your Desktop user, try running your application as a Windows service instead as explained in the documentation.

**Comment by jvinayagam** [comment]
Thanks for your response. really appreciate that
more detail:

we can ran the service in windows server as java -jar documents4j-server-standalone-shaded.jar http://server:9998


whats happening in our server. every 1 hours system is logging off. so that service is lost. how we can resolve this

 so that i have put execute through windows task scheduler on start up with user is logged on. its not working when i select user is logged or logged not in the task schduler
i have created the folder too
C:\Windows\System32\config\systemprofile\Desktop






in our  client java code

URL url = new URL("http://server:9998");
		HttpURLConnection conn = (HttpURLConnection) url.openConnection();
		conn.setDoOutput(true);
		conn.setRequestMethod("POST");
		conn.setRequestProperty("Content-Type", "application/vnd.com.documents4j.any-msword");
		conn.setRequestProperty("Accept", "application/pdf");
		conn.setRequestProperty("Converter-Job-Priority", "1000");

		OutputStream os = conn.getOutputStream();

		os.write(docxBytesArray);
		os.flush();

		if (conn.getResponseCode() != HttpURLConnection.HTTP_OK) {
			throw new RuntimeException("Failed : HTTP error code : " + conn.getResponseCode());
		}

		/*
		 * BufferedReader br = new BufferedReader(new InputStreamReader(
		 * (conn.getInputStream()))); FileWriter ostream = new
		 * FileWriter("C:/tmp/test.pdf"); BufferedWriter out = new
		 * BufferedWriter(ostream); String output;
		 * System.out.println("Output from Server .... \n"); while ((output =
		 * br.readLine()) != null) { System.out.println(output);
		 * out.write(output+"\n"); }
		 */

		InputStream in = conn.getInputStream();
		// OutputStream out = new FileOutputStream("C:/tmp/gillgood.pdf");
		ByteArrayOutputStream pdfbuffer = new ByteArrayOutputStream();
		byte[] buffer = new byte[1024];
		int len;
		while ((len = in.read(buffer)) != -1) {
			// System.out.println(buffer);
			// out.write(buffer, 0, len);
			pdfbuffer.write(buffer, 0, len);

		}
		// br.close();
		// out.close();
		os.close();
		conn.disconnect();

		return pdfbuffer.toByteArray();


how we can resolve the issue without login  to run as service in window machine?

**Comment by raphw** [comment]
I am not sure if this is really anything you can solve in a Java application. I assume that the desktop user times. This is what the service interface is used for. Maybe it is possible to activate some sort of server mode also for regular users but this is nothing I know anything about.

**Comment by vinayajo** [comment]
Thanks.  is it possible to create a window service ( running jar file in the background) without logon to windows server ?

And also in order to run these jar file as service? is it mandatory to logon on to run?

**Comment by raphw** [comment]
I think the server must be running what is all required to run a service. This is however not my field of expertise.

**Comment by jvinayagam** [comment]
is it possible to run as service in redhat linux without microsoft office components?.

**Comment by jvinayagam** [comment]
Thanks service is running problem it requires logon. is it possible to run as service in redhat linux without microsoft office components?.

**Comment by raphw** [comment]
No, it is only possible to run on a Windows Server that supports .NET and has Office installed.
